### ISSUE TITLE
Scripting

### ISSUE THREAD

**Issue by haesleinhuepf** [maarzt/imglib2-labkit#46]
Hi @maarzt,

I just trained a model successfully (it looks really great!) Is there a way for executing this model on 1000 images? Idealy via ImageJ macro as I have pre- and post-processing steps I would like to call.

Thanks a lot!

Cheers,
Robert

**Comment by haesleinhuepf** [comment]
Hey @maarzt ,

are there any news on the scripting side? A user in the forum might be interested in trying out gpu-labkit from Macro!

https://forum.image.sc/t/automating-trainable-weka-segmentation-on-clij2/40632

Let me know if I can help in making a macro interface 🙃

Cheers,
Robert

**Comment by maarzt** [comment]
No, there's currently no interface for scripting Labkit with imagej macro.
But that's something I would like to implement soon.
I would be happy if anybody starts coding a prototype ;)

**Comment by haesleinhuepf** [comment]
> I would be happy if anybody starts coding a prototype ;)

Can you point me to code snippets where one can see how a model is loaded from disc and how it's applied to a Dataset? I'd give it a go then. ;-)

**Comment by maarzt** [comment]
Hi @haesleinhuepf,

The segmentation algorithm is all implemented in github.com/maarzt/imglib2-trainable-segmentation.
This little, snippet should do the trick of loading a trained segmentation model, and applying it to an image:
```java
package net.imglib2.trainable_segmentation.classification;

import bdv.util.BdvFunctions;
import net.imglib2.RandomAccessibleInterval;
import net.imglib2.trainable_segmentation.gson.GsonUtils;
import net.imglib2.type.numeric.integer.UnsignedByteType;
import net.imglib2.type.numeric.real.FloatType;
import org.scijava.Context;

public class SegmenterDemo {

	public static void main(String... args) {
		String filename = "your.classifier";
		RandomAccessibleInterval<FloatType> image = null; // open your image.
		Segmenter segmenter = Segmenter.fromJson(new Context(), GsonUtils.read(filename));
		RandomAccessibleInterval<UnsignedByteType> result = segmenter.segment(image);
		BdvFunctions.show(result, "segmentation").setDisplayRange(0, 2);
	}
}
```

**Comment by maarzt** [comment]
You would need to call ```segmenter.setUseGpu(true)``` to enable GPU calculation.

**Comment by maarzt** [comment]
This is fixed by #50
